api/src/main/java/io/minio/MinioClient.java

```diff
@@ -1884,6 +1888,156 @@ public void removeObject(String bucketName, String objectName)
   }
 
 
+  private List<DeleteError> removeObject(String bucketName, List<DeleteObject> objectList)
+    throws InvalidBucketNameException, NoSuchAlgorithmException, InsufficientDataException, IOException,
+           InvalidKeyException, NoResponseException, XmlPullParserException, ErrorResponseException,
+           InternalException {
+    Map<String,String> queryParamMap = new HashMap<>();
+    queryParamMap.put("delete", "");
+
+    DeleteRequest request = new DeleteRequest(objectList);
+    HttpResponse response = executePost(bucketName, null, null, queryParamMap, request);
+
+    String bodyContent = "";
+    Scanner scanner = new Scanner(response.body().charStream());
+    try {
+      // read entire body stream to string.
+      scanner.useDelimiter("\\A");
+      if (scanner.hasNext()) {
+        bodyContent = scanner.next();
+      }
+    } finally {
+      response.body().close();
+      scanner.close();
+    }
+
+    List<DeleteError> errorList = null;
+
+    bodyContent = bodyContent.trim();
+    DeleteError error = new DeleteError(new StringReader(bodyContent));
```

Comment:
Can you also add a comment for this, to say that we need sometimes to parse the body twice since the xml tree structure is not the same in some circumstances.